### PR TITLE
feat: adding validation through the subscription

### PR DESCRIPTION
**Pull request by Skarlso**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue # (issue)
- Closes # (issue)
- Fixes # (issue)
> Remove if not applicable

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🙅 no documentation needed

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Skarlso** [comment]
https://github.com/open-component-model/ocm-controller/pull/316

File: controllers/productdeployment_controller.go [hunk 1]
```diff
@@ -310,6 +310,15 @@ func (r *ProductDeploymentReconciler) createOrUpdatedValuesConfigMap(
 }
 
 func (r *ProductDeploymentReconciler) createOrUpdateComponentVersion(ctx context.Context, obj *v1alpha1.ProductDeployment) error {
+	signature := make([]ocmv1alpha1.Signature, 0, len(obj.Spec.Verify))
+	for _, s := range obj.Spec.Verify {
+		signature = append(signature, ocmv1alpha1.Signature{
+			Name:          s.Name,
+			PublicKey:     ocmv1alpha1.SecretRef{SecretRef: s.PublicKey.SecretRef},
```
Comment (Skarlso) [inline controllers/productdeployment_controller.go:317]: Ah freck. This should have been a pointer in replication controller. :/

**Review by souleb — APPROVED** [review]

File: controllers/productdeployment_controller.go (hunk 1 above)
Comment (Skarlso) [reply controllers/productdeployment_controller.go:317]: Going to release this and use it in git-controller then update this part here. 

https://github.com/open-component-model/ocm-controller/pull/325

Comment (Skarlso) [reply controllers/productdeployment_controller.go:317]: Merged, and replication controller update is running. I'll be applying the new version shortly.

**Pull request merged**